src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += ',';
+        continue;
+      }
+
+      if (i === this.length - 1) {
+        result += this[i];
+        break;
```

Comment:
```suggestion
      if (i < this.length - 1) {
        result += separator;
```
You may add separator always after each element while it's not the last element. In this case, nothing will be added